File: Daily/sbgo/Leetcode581-最短无序连续子数组.go
package sbgo

import "sort"

func findUnsortedSubarray(nums []int) int {
	n := len(nums)
	cp := make([]int, n)
	copy(cp, nums)
	sort.Ints(nums)
	l, r := -1, 0
	for i := 0; i < n; i++ {
		if l == -1 && nums[i] != cp[i] {
			l = i
		} else if l != -1 && nums[i] != cp[i] {
			r = i
		}
	}
	if l == -1 {
		return 0
	}
	return r - l + 1
}

File: Leetcode/PY/Leetcode581-最短无序连续子数组.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        cp = nums[:]
        nums.sort()
        n = len(nums)
        l,r = -1,0
        for i in range(n):
            if l==-1 and nums[i] != cp[i]:
                l = i
            elif l != -1 and nums[i] != cp[i]:
                r = i

        if l == -1:
            return 0
        return r - l + 1